File: backend/src/NHLStats.Domain/Entities/UserMatch.cs
namespace NHLStats.Domain.Entities;

public class UserMatch
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int MatchId { get; set; }
    public int SeasonId { get; set; }

    public User? User { get; set; }
    public Match? Match { get; set; }
    public Season? Season { get; set; }
    public ICollection<UserMatchPoint> Points { get; set; } = new List<UserMatchPoint>();
    public ICollection<UserMatchGoal> Goals { get; set; } = new List<UserMatchGoal>();
    public ICollection<UserMatchPenalty> Penalties { get; set; } = new List<UserMatchPenalty>();
}

File: frontend/src/types/userMatch.ts
import type { PointType } from './pointReason'

export type GoalType = 'Regular' | 'PowerPlay' | 'ShortHanded'

export interface UserMatch {
    id: number
    userId: number
    userName: string | null
    matchId: number | null
    seasonId: number
}

export interface UserMatchPoint {
    id: number
    userMatchId: number
    pointReasonId: number
    pointReasonName: string | null
    pointType: PointType
    count: number
}

export interface UserMatchGoal {
    id: number
    userMatchId: number
    rosterPlayerId: number
    playerFirstName: string | null
    playerSurname: string | null
    count: number
    goalType: GoalType
}

export interface UserMatchPenalty {
    id: number
    userMatchId: number
    rosterPlayerId: number
    playerFirstName: string | null
    playerSurname: string | null
    count: number
}

export interface CreateUserMatchPointDto {
    pointReasonId: number
    count: number
}

export interface CreateUserMatchGoalDto {
    rosterPlayerId: number
    count: number
    goalType: GoalType
}

export interface CreateUserMatchPenaltyDto {
    rosterPlayerId: number
    count: number
}
